xlive/Blam/Engine/rasterizer/rasterizer_globals.h

```diff
@@ -1,5 +1,119 @@
 #pragma once
+#include "game/game_globals.h"
 
+/* enums */
+
+enum e_rasterizer_window_mode : uint32
+{
+	_rasterizer_window_mode_real_fullscreen = 0,
+	_rasterizer_window_mode_windowed = 1,
+	_rasterizer_window_mode_funky_fullscreen = 2
+};
+
+enum e_display_type : uint8
+{
+	_display_type_widescreen = 0,
+	_display_type_4_by_3 = 1
+};
+
+/* structures */
+
+struct s_rasterizer_globals_display_parameters
+{
+	uint64 frame_presented_count;
+	int64 pad_2;
+	int64 field_8;
+	int64 pad_3;
+	int16 refresh_rate;
+	int16 pad;
+	D3DFORMAT backbuffer_format;
+	D3DFORMAT depthstencil_format;
+	int16 font_width;
+	int16 font_height;
+	e_display_type display_type;
+	int8 unused_window_mode;
+	int8 pad_1[2];
+	e_rasterizer_window_mode window_mode;
+};
+ASSERT_STRUCT_SIZE(s_rasterizer_globals_display_parameters, 56);
+
+struct s_rasterizer_globals_clipping_parameters
+{
+	uint32 field_0[3];
+	real32 z_near;
+	real32 z_far;
+	real32 depth_near;
+	real32 depth_far;
+};
+ASSERT_STRUCT_SIZE(s_rasterizer_globals_clipping_parameters, 28);
+
+struct s_rasterizer_globals_bitmaps
+{
+	datum bitmap_2d_unused;
+	datum bitmap_3d_unused;
+	datum bitmap_cubemap_unused;
+	bool field_90;
+	bool field_91;
+	int8 pad_2[2];
+	uint32 textures_staged_width[4];
+	uint32 textures_staged_height[4];
+};
+ASSERT_STRUCT_SIZE(s_rasterizer_globals_bitmaps, 48);
+
+struct s_rasterizer_fullscreen_parameters
+{
+	real32 gamma;
+	real32 brightness;
+	real32 field_8;	// unused
+	real32 field_C;	// unused
+};
+ASSERT_STRUCT_SIZE(s_rasterizer_fullscreen_parameters, 16);
+
+// TODO: Finish this, some data above might be included in it
+struct s_rasterizer_globals
```

Comment:
by the same logic, this should also be removed?

Reply:
This is a struct that exists in the original game.
After taking another look, the globals above it are related to shader submit logic.